### ISSUE TITLE
Socks proxy hides good errors when the site doesn't exist

### ISSUE THREAD

**Issue by pbiggar** [darklang/dark#4005]
When a user makes a request to a site which doesn't exist, the error looks like:

`<ErrorRail: Error "The request was aborted. The response ended prematurely.">`

However, if we disable the socks proxy, we get a much better error: 

`<ErrorRail: Error "Name or service not known (the.customer.url.com:443) Name or service not known">`

In OCaml, the error was:

`<ErrorRail: Error "Couldn't connect to server">`

With curl:

```
curl --socks5-hostname localhost:1080 http://thenonexistingurlforsure.com
curl: (7) Failed to receive SOCKS5 connect request ack.
```

**Comment by pbiggar** [comment]
I'm not sure the httpclient can do any better here. It doesn't do DNS itself, but instead goes through the proxy, and I think the proxy just cuts it off.

One option is to change the proxy and see if that helps. I've found https://github.com/serjs/socks5-server which is an option. I've also reached out to GCP to see if there's something in their product suite that does this.

Another option is just that when we get an IO exception, to cover it up with a nicer error (we already log the error that we get).
